pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/Policies.java

```diff
@@ -18,15 +18,52 @@
  */
 package org.apache.pulsar.common.policies.data;
 
+import java.lang.reflect.Type;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
 import com.google.common.base.MoreObjects;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
+import com.google.common.reflect.TypeToken;
 
 public class Policies {
+    public enum PolicyProperty {
```

Comment:
We should define this enum outside the scope of `Policies` class, in its own file.